python/cudf/cudf/core/indexing.py

```diff
@@ -168,13 +168,18 @@ def _loc_to_iloc(self, arg):
         from cudf.core.series import Series
 
         if is_scalar(arg):
-            try:
-                found_index = self._sr.index._values.find_first_value(
-                    arg, closest=False
-                )
+            if isinstance(arg, int):
+                found_index = arg
                 return found_index
-            except (TypeError, KeyError, IndexError, ValueError):
-                raise KeyError("label scalar is out of bound")
+            
+            if isinstance(arg, str):
```

Comment:
Index values aren't necessary strings, also `arg` needs to handle `cudf.Scalar` objects as well.